<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenVPNProp</class>
 <widget class="QWidget" name="OpenVPNProp">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>495</width>
    <height>422</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="textLabel3">
       <property name="text">
        <string>Gateway:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="buddy">
        <cstring>gateway</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="gateway">
       <property name="toolTip">
        <string>Remote gateways(s), with optional port and protocol (e.g. ovpn.corp.com:1234:tcp).
You can specify multiple hosts for redundancy (use commas or spaces as delimiters).</string>
       </property>
       <property name="clearButtonEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelConnectionType">
       <property name="text">
        <string>Connection type:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="buddy">
        <cstring>cmbConnectionType</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="cmbConnectionType">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Select an authentication mode.</string>
       </property>
       <item>
        <property name="text">
         <string>Certificates (TLS)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Static Key</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Password</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Password with Certificates (TLS)</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page">
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="textLabel2">
         <property name="text">
          <string>CA Certificate:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509CaFile</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KUrlRequester" name="x509CaFile">
         <property name="toolTip">
          <string>Certificate authority (CA) file in .pem format.</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="textLabel3_2">
         <property name="text">
          <string>User Certificate:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509Cert</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="KUrlRequester" name="x509Cert">
         <property name="toolTip">
          <string>Local peer's signed certificate in .pem format (signed by CA of CA Certificate).</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="textLabel4">
         <property name="text">
          <string>Private Key:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509Key</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="KUrlRequester" name="x509Key">
         <property name="toolTip">
          <string>Local peer's private key in .pem format.</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Private Key Password:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="PasswordField" name="x509KeyPassword">
           <property name="toolTip">
            <string>Password for private key.</string>
           </property>
           <property name="passwordModeEnabled" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="4" column="0" colspan="2">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>347</width>
           <height>206</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_2">
      <layout class="QFormLayout" name="formLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="textLabel5">
         <property name="text">
          <string>Static Key:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>pskSharedKey</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KUrlRequester" name="pskSharedKey">
         <property name="toolTip">
          <string>Pre-shared file for Static Key encryption mode (non-TLS).</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.key</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Key Direction:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="cmbKeyDirection">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string>Direction for Static Key encryption mode (non-TLS).</string>
         </property>
         <property name="whatsThis">
          <string>If key direction is used, it must be the opposite of that used on the VPN peer.</string>
         </property>
         <item>
          <property name="text">
           <string comment="No &quot;Key Direction&quot;">None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="textLabel6">
         <property name="text">
          <string>Local IP Address:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>pskLocalIp</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="pskLocalIp">
         <property name="toolTip">
          <string>IP address of the remote VPN endpoint.</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="textLabel7">
         <property name="text">
          <string>Remote IP Address:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>pskRemoteIp</cstring>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QLineEdit" name="pskRemoteIp">
         <property name="toolTip">
          <string>IP address of the local VPN endpoint.</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="2">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="lblKeyDirection">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <italic>true</italic>
          </font>
         </property>
         <property name="text">
          <string>If key direction is used, it must be the opposite of that used on the VPN peer. For example, if the peer uses '1', this connection must use '0'. If you are unsure what value to use, contact your system administrator.</string>
         </property>
         <property name="textFormat">
          <enum>Qt::AutoText</enum>
         </property>
         <property name="scaledContents">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_3">
      <layout class="QFormLayout" name="formLayout_4">
       <item row="0" column="0">
        <widget class="QLabel" name="textLabel8">
         <property name="text">
          <string>CA Certificate:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>passCaFile</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KUrlRequester" name="passCaFile">
         <property name="toolTip">
          <string>Certificate authority (CA) file in .pem format.</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="textLabel9">
         <property name="text">
          <string comment="like in Username for athentication">Username:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>passUserName</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="passUserName">
         <property name="toolTip">
          <string>Username passed to OpenVPN when prompted for it.</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Password:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <widget class="PasswordField" name="passPassword">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>Password passed to OpenVPN when prompted for it.</string>
           </property>
           <property name="passwordModeEnabled" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0" colspan="2">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_4">
      <layout class="QFormLayout" name="formLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="textLabel2_2">
         <property name="text">
          <string>CA Certificate:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509PassCaFile</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="KUrlRequester" name="x509PassCaFile">
         <property name="toolTip">
          <string>Certificate authority (CA) file in .pem format.</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="textLabel3_2_2">
         <property name="text">
          <string>User Certificate:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509PassCert</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="KUrlRequester" name="x509PassCert">
         <property name="toolTip">
          <string>Local peer's signed certificate in .pem format (signed by CA of CA Certificate).</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="textLabel4_2">
         <property name="text">
          <string>Private Key:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509PassKey</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="KUrlRequester" name="x509PassKey">
         <property name="toolTip">
          <string>Local peer's private key in .pem format.</string>
         </property>
         <property name="nameFilters">
          <StringList>
           <string notr="true">*.pem *.crt *.key *.cer *.p12</string>
          </StringList>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Private Key Password:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_12">
         <item>
          <widget class="PasswordField" name="x509PassKeyPassword">
           <property name="toolTip">
            <string>Password for private key.</string>
           </property>
           <property name="passwordModeEnabled" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="textLabel9_2">
         <property name="text">
          <string comment="like in Username for athentication">Username:</string>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="buddy">
          <cstring>x509PassUsername</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLineEdit" name="x509PassUsername">
         <property name="toolTip">
          <string>Username passed to OpenVPN when prompted for it.</string>
         </property>
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Password:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_13">
         <item>
          <widget class="PasswordField" name="x509PassPassword">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="toolTip">
            <string>Password passed to OpenVPN when prompted for it.</string>
           </property>
           <property name="passwordModeEnabled" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="6" column="0" colspan="2">
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnAdvanced">
       <property name="text">
        <string>Advanced…</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QWidget</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>PasswordField</class>
   <extends>QLineEdit</extends>
   <header>passwordfield.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>gateway</tabstop>
  <tabstop>cmbConnectionType</tabstop>
  <tabstop>x509CaFile</tabstop>
  <tabstop>x509Cert</tabstop>
  <tabstop>x509Key</tabstop>
  <tabstop>x509KeyPassword</tabstop>
  <tabstop>pskSharedKey</tabstop>
  <tabstop>cmbKeyDirection</tabstop>
  <tabstop>pskLocalIp</tabstop>
  <tabstop>pskRemoteIp</tabstop>
  <tabstop>passCaFile</tabstop>
  <tabstop>passUserName</tabstop>
  <tabstop>passPassword</tabstop>
  <tabstop>x509PassCaFile</tabstop>
  <tabstop>x509PassCert</tabstop>
  <tabstop>x509PassKey</tabstop>
  <tabstop>x509PassKeyPassword</tabstop>
  <tabstop>x509PassUsername</tabstop>
  <tabstop>x509PassPassword</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cmbConnectionType</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>584</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
